corehq/tabs/uitab.py

```diff
@@ -66,12 +66,9 @@ def request_path(self):
         return self._request.get_full_path()
 
     @property
-    def restrict_access_by_location(self):
-        """Is this a web user who can't access project-wide data?"""
-        return (
-            self.couch_user and self.domain
-            and not self.couch_user.has_permission(self.domain, 'access_all_locations')
-        )
+    def can_access_all_locations(self):
+        """Is this a web user who can access project-wide data?"""
+        return getattr(self._request, 'can_access_all_locations', True)
```

Comment:
Sorta.  The assumption is that if this isn't set, it's because this is some special case where it didn't go through middleware.  The middleware that sets this also defaults to True if it can't figure out the user's permissions, such as if the user isn't a member of the domain, or if the request doesn't have a domain.  I'm intentionally being pretty defensive - the idea is that certain users are explicitly marked as restricted when accessing pages associated with a domain.  If the users aren't marked, or we can't figure out whether they're marked, it falls back to the current behavior (which may be, for example, raising an exception in `login_and_domain_required`).

I do agree in principle with being stricter about this sort of thing, and I think this is important to get right, so I'd welcome suggestions about how to do this safely without failing on a bunch of obscure edge cases.